### PR TITLE
Ensure e2e HTTP limit args are passed all the way through to the nodes

### PR DESCRIPTION
**Pull request by eddyashton**
Because of the multiple paths we have to construct the node interfaces, we weren't actually passing the CLI-controlled `args` in several situations.

Before:

```
$ ../tests/sandbox/sandbox.sh --max-http-body-size 1 --max-http-header-size 2 --max-http-headers-count 3 -p libjs_generic
...

$ cat workspace/sandbox_0/0.config.json
...
  ... "http_configuration": {"max_body_size": "1048576", "max_header_size": "16384", "max_headers_count": 256} ...
...
```

After:
```
$ ../tests/sandbox/sandbox.sh --max-http-body-size 1 --max-http-header-size 2 --max-http-headers-count 3 -p libjs_generic
...

$ cat workspace/sandbox_0/0.config.json
...
        "http_configuration": {
          "max_body_size": "1",
          "max_header_size": "2",
          "max_headers_count": 3
        },
...
```

This fixes the initial bug report, when networks are started via `sandbox.sh`, and hopefully sufficient for existing tests. It doesn't change the config used if we call `create_node`, because many of the paths there don't have access to the (CLI-populated) `args`. A neater solution would probably be to try and make those CLI args a little more global instead of passing them around, but that's a much bigger refactor.

### CODE REVIEW

File: tests/infra/remote.py [hunk 1]
```diff
@@ -791,7 +791,10 @@ def __init__(
             exe_files += [config_file]
 
             with open(config_file, "w", encoding="utf-8") as f:
-                f.write(output)
+                # Parse and re-emit output to produce consistently formatted (indented) JSON.
+                # This will also ensure the render produced valid JSON
+                j = json.loads(output)
+                json.dump(j, f, indent=2)
```
Comment (eddyashton) [inline tests/infra/remote.py:797]: This is a kind-of-unrelated driveby change, but I think a useful QoL improvement. The `jinja`-rendered configs have some quirky formatting due to constraints in the original template, which make them harder to read by hand. This is a wastefully full re-encoding, but negligible in the startup time, and produces consistent formatting.

```
    "rpc_interfaces": {"primary_rpc_interface": {"app_protocol": "HTTP1", "bind_address": "127.0.0.1:8000", "endorsement": {"authority": "Service"}, "http_configuration": {"max_body_size": "1048576", "max_header_size": "16384", "max_headers_count": 256}, "max_open_sessions_hard": 1010, "max_open_sessions_soft": 1000, "protocol": "tcp", "published_address": "127.0.0.1:8000"}}
...
      "members": [{"certificate_file": "member0_cert.pem", "data_json_file": null, "encryption_public_key_file": "member0_enc_pubk.pem"}],
```
->
```
    "rpc_interfaces": {
      "primary_rpc_interface": {
        "app_protocol": "HTTP1",
        "bind_address": "127.0.0.1:8000",
        "endorsement": {
          "authority": "Service"
        },
        "forwarding_timeout_ms": 3000,
        "http_configuration": {
          "max_body_size": "1",
          "max_header_size": "2",
          "max_headers_count": 3
        },
        "max_open_sessions_hard": 1010,
        "max_open_sessions_soft": 1000,
        "protocol": "tcp",
        "published_address": "127.0.0.1:8000"
      }
    }
...
      "members": [
        {
          "certificate_file": "member0_cert.pem",
          "data_json_file": null,
          "encryption_public_key_file": "member0_enc_pubk.pem"
        }
      ],
```

Or more usefully
```
$ grep "max_header_size" workspace/sandbox_0/0.config.json
    "rpc_interfaces": {"primary_rpc_interface": {"app_protocol": "HTTP1", "bind_address": "127.0.0.1:8000", "endorsement": {"authority": "Service"}, "http_configuration": {"max_body_size": "1048576", "max_header_size": "16384", "max_headers_count": 256}, "max_open_sessions_hard": 1010, "max_open_sessions_soft": 1000, "protocol": "tcp", "published_address": "127.0.0.1:8000"}}
```
->
```
$ grep "max_header_size" workspace/sandbox_0/0.config.json
          "max_header_size": "2",
```

**Review by jumaffre — APPROVED** [review]

**Review by achamayou — APPROVED** [review]

**Comment by eddyashton** [comment]
## 💔 All backports failed

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|release/3.x|Conflict resolution was aborted by the user|

### Manual backport
To create the backport manually run:
```
backport --pr 5348
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by eddyashton** [comment]
## 💔 All backports failed

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|release/3.x|An unhandled error occurred. Please see the logs for details|

### Manual backport
To create the backport manually run:
```
backport --pr 5348
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by eddyashton** [comment]
Don't fully understand why backport CLI struggled so much with 3.x, so I've created that PR by hand: https://github.com/microsoft/CCF/pull/5369

**Pull request merged**